doc/reference/api.signals/BaseSignal.rst

```diff
@@ -0,0 +1,4 @@
+.. autoclass:: hyperspy.api.signals.BaseSignal
```

Comment:
all `api,signals` pages are missing the `:mod:` header and e.g. in a search appear as pages without title

Reply:
Indeed, I have various approach without much success. Adding a section has the disadvantage of making the TOC on the right hand useless, as only the top one is displayed.
It may be a sphinx bug that no title is being pick up for this page...

I am giving up on that one! If someone who understand sphinx and the autodoc extension can have a go to fix it or understand where the issue is, that would be nice! :)

Reply:
Well you have e.g.

```
:mod:`hyperspy.api.model.components2D`
--------------------------------------

.. autoclass:: hyperspy.api.model.components2D.Expression
   :noindex:
```

So why not do the same here?
```
:mod:`hyperspy.api.signals.BaseSignal`
--------------------------------------

.. autoclass:: hyperspy.api.signals.BaseSignal
```

Could that correct the titles?

Reply:
This is one of the approach that I have tried and by doing this, we get the title but lose the "TOC" of the current page on the right hand side of the page. I find this TOC quite useful to give an overview of the content of the page and for the moment, I prefer to keep it in favour of the title of the page.

Reply:
I have got it sorted: the configuration settings (`"show_toc_level": 2`) was missing.